cli/src/cmd/forge/script/mod.rs

```diff
@@ -410,6 +416,72 @@ impl ScriptArgs {
         };
         Ok((func.clone(), data))
     }
+
+    /// Checks if the transaction is a deployment with a size above the `CONTRACT_MAX_SIZE`.
+    ///
+    /// If `self.broadcast` is enabled, it asks confirmation of the user. Otherwise, it just warns
+    /// the user.
+    fn check_contract_sizes(
+        &self,
+        transactions: Option<&VecDeque<TypedTransaction>>,
+        known_contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
+    ) -> eyre::Result<()> {
+        if let Some(txes) = transactions {
+            for tx in txes {
+                if let Some(data) = tx.data() {
+                    if data.len() > CONTRACT_MAX_SIZE {
+                        let mut offset = 0;
+
+                        // Find if it's a CREATE or CREATE2. Otherwise, skip transaction.
+                        if let Some(NameOrAddress::Address(to)) = tx.to() {
+                            if *to == DEFAULT_CREATE2_DEPLOYER {
+                                // Size of the salt prefix.
+                                offset = 32;
+                            }
+                        } else if tx.to().is_some() {
+                            continue
+                        }
+
+                        // Find artifact with a deployment code same as the data.
+                        if let Some((artifact, bytecode)) =
+                            known_contracts.iter().find(|(_, bytecode)| {
+                                bytecode
+                                    .bytecode
+                                    .object
+                                    .as_bytes()
+                                    .expect("Code should have been linked before.") ==
+                                    &data[offset..]
+                            })
+                        {
+                            // Find the deployed code size of the artifact.
+                            if let Some(deployed_bytecode) = &bytecode.deployed_bytecode.bytecode {
+                                let deployment_size = deployed_bytecode.object.bytes_len();
+
+                                if deployment_size > CONTRACT_MAX_SIZE {
+                                    println!(
+                                        "{}",
+                                        Paint::red(format!(
+                                            "`{}` is above the contract size limit ({} vs {}).",
+                                            artifact.name, deployment_size, CONTRACT_MAX_SIZE
+                                        ))
+                                    );
+
+                                    if self.broadcast &&
+                                        !Confirm::new()
+                                            .with_prompt("Do you wish to continue?".to_string())
+                                            .interact()?
+                                    {
+                                        eyre::bail!("User canceled the script.");
```

Comment:
not a fan of stdin here

is there a good reason why we want stdin support here

Reply:
It could just fail... but I was thinking of giving the option to the user, in case he's still in the development phase (say running tests against anvil, which doesn't check contract sizes), and he's not too worried about it for now. Not a very strong opinion though.